SEGV when trying to "zar find" on indexed "query" field

Repro is with `zar` commit `ed46b3d`. Using the zar demo steps:

```
# zar index -d ~/logs query
/Users/phil/logs/20170324/1490381125.199124.zng: creating index /Users/phil/logs/20170324/1490381125.199124.zng.zar/zdx:field:query
/Users/phil/logs/20170324/1490383711.041045.zng: creating index /Users/phil/logs/20170324/1490383711.041045.zng.zar/zdx:field:query
...

# zar find -d ~/logs query="soap.com"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x1153cd3]

goroutine 1 [running]:
github.com/brimsec/zq/archive.Search(0xc0000ae030, 0x2f, 0x11e51c0, 0xc0000a82e0, 0x7ffeefbffa0b, 0x8, 0xc0000f38a0, 0x103a3fc, 0xc000000180)
	/Users/phil/work/zq/archive/finder.go:66 +0x203
github.com/brimsec/zq/archive.Find.func1(0xc0000ae030, 0x2f, 0x11e6bc0, 0xc00009d5f0, 0x0, 0x0, 0x15, 0xc0000f39a0)
	/Users/phil/work/zq/archive/finder.go:38 +0x2ce
path/filepath.walk(0xc0000ae030, 0x2f, 0x11e6bc0, 0xc00009d5f0, 0xc0000f3c28, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:360 +0x425
path/filepath.walk(0xc0000de0a0, 0x19, 0x11e6bc0, 0xc00009d520, 0xc0000f3c28, 0x0, 0x0)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:384 +0x2ff
path/filepath.walk(0x7ffeefbff9f4, 0x10, 0x11e6bc0, 0xc00009d450, 0xc0000f3c28, 0x0, 0xc)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:384 +0x2ff
path/filepath.Walk(0x7ffeefbff9f4, 0x10, 0xc0000f3c28, 0x103b1be, 0xc0000f3c28)
	/usr/local/opt/go/libexec/src/path/filepath/path.go:406 +0xff
github.com/brimsec/zq/archive.Find(0x7ffeefbff9f4, 0x10, 0x11e51c0, 0xc0000a82e0, 0x7ffeefbffa0b, 0x8, 0xc00008e0c0, 0x2, 0x2)
	/Users/phil/work/zq/archive/finder.go:23 +0x9f
github.com/brimsec/zq/cmd/zar/find.(*Command).Run(0xc0000a8260, 0xc0000b6040, 0x1, 0x1, 0xc0000b6040, 0x1)
	/Users/phil/work/zq/cmd/zar/find/command.go:75 +0x331
github.com/mccanne/charm.(*instance).run(0xc0000a8280, 0xc0000b6020, 0x3, 0x3, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x28c
github.com/mccanne/charm.(*Spec).Exec(0x1301b40, 0x11e3de0, 0xc000098c10, 0xc0000b6020, 0x3, 0x3, 0x4, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:67 +0x92
github.com/mccanne/charm.(*instance).run(0xc0000a8240, 0xc0000b6010, 0x4, 0x4, 0x0, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:61 +0x41d
github.com/mccanne/charm.(*Spec).ExecRoot(0x1301c80, 0xc0000b6010, 0x4, 0x4, 0xc0000f3f78, 0x1005fbf, 0xc00008e058, 0x0)
	/Users/phil/.go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
	/Users/phil/work/zq/cmd/zar/main.go:19 +0x78
```

--- a/archive/finder.go
+++ b/archive/finder.go
@@ -63,6 +63,11 @@ func Search(path string, rule Rule, pattern string) (bool, error) {
 		}
 		return false, err
 	}
+	defer finder.Close()
+	if keyType == nil {
+		// This happens when an index exists but is empty.
+		return false, nil
+	}
 	keyBytes, err := keyType.Parse([]byte(pattern))
 	if err != nil {
 		return false, fmt.Errorf("%s: %w", finder.Path(), err)

--- a/cmd/zdx/lookup/command.go
+++ b/cmd/zdx/lookup/command.go
@@ -74,10 +74,10 @@ func (c *LookupCommand) Run(args []string) error {
 	if err != nil {
 		return err
 	}
+	defer finder.Close()
 	if keyType == nil {
 		return fmt.Errorf("%s: index is empty", path)
 	}
-	defer finder.Close()
 	// XXX Parse doesn't work yet for record values, but everything else
 	// is ready to go to use records and index keys
 	key, err := keyType.Parse([]byte(c.key))

--- a/tests/suite/zdx/empty.yaml
+++ b/tests/suite/zdx/empty.yaml
@@ -1,0 +1,8 @@
+script: |
+  echo "" > empty.zng
+  zdx lookup -t -k none empty
+
+outputs:
+  - name: stderr
+    regexp: |
+      empty: index is empty

--- a/tests/suite/zdx/empty.yaml
+++ b/tests/suite/zdx/empty.yaml
@@ -1,5 +1,5 @@
 script: |
-  echo "" > empty.zng
+  echo -n "" > empty.zng
   zdx lookup -t -k none empty
 
 outputs:

--- a/zdx/zdx_test.go
+++ b/zdx/zdx_test.go
@@ -108,6 +108,8 @@ func TestSearch(t *testing.T) {
 	if !bytes.Equal(value, value2) {
 		t.Error("key lookup failed")
 	}
+	err = finder.Close()
+	require.NoError(t, err)
 }
 
 func TestZdx(t *testing.T) {

